python/pyspark/testing/utils.py

```diff
@@ -152,7 +153,7 @@ def close(self):
 def search_jar(project_relative_path, sbt_jar_name_prefix, mvn_jar_name_prefix):
     # Note that 'sbt_jar_name_prefix' and 'mvn_jar_name_prefix' are used since the prefix can
     # vary for SBT or Maven specifically. See also SPARK-26856
-    project_full_path = os.path.join(os.environ["SPARK_HOME"], project_relative_path)
+    project_full_path = os.path.join(_find_spark_home(), project_relative_path)
```

Comment:
I think we can just use `SPARK_HOME`. SPARK_HOME should be set before importing this in our testing script anyway.